helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementService.java

```diff
@@ -528,10 +560,7 @@ private List<String> filterInstancesForNextCheck(
       String instance = entry.getKey();
       try {
         StoppableCheck stoppableCheck = entry.getValue().get();
-        if (!stoppableCheck.isStoppable()) {
-          // put the check result of the failed-to-stop instances
-          addStoppableCheck(finalStoppableCheckByInstance, instance, stoppableCheck);
-        }
+        addStoppableCheck(finalStoppableCheckByInstance, instance, stoppableCheck);
```

Comment:
You can add all the stoppableChecks because isStoppable only checks for checks that are failed.

This does not affect if the instances are filtered out.

That logic remains the same here: https://github.com/apache/helix/pull/2641/files#diff-423574b747e8866eced6d0352a1a8ac244158359f4ecaefca3e0a57af68203c6R567